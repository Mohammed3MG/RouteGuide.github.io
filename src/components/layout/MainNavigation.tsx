import { Container, Navbar, Nav } from "react-bootstrap";
import { Link } from "react-router-dom";
import logo from "../../assets/img/logo.webp"

const MainNavigation = () => {
  return (
    <Navbar data-testid="main-navigation" bg="light" expand="lg">
      <Container>
        <Link to="/" className="navbar-brand">
          <span role="img" aria-label="map">
            <img src={logo} height={25} />
          </span>{" "}
         Travel Route guide
        </Link>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            <Link to="/home" className="nav-link">
              Home
            </Link>
            <Link to="/cities" className="nav-link">
              Cities
            </Link>
           
            <Link to="/result" className="nav-link">
              Result
            </Link>
          </Nav>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  );
};

export default MainNavigation;
