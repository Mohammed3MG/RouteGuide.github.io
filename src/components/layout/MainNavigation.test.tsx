// MainNavigation.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter } from 'react-router-dom'; // MemoryRouter is used for testing with react-router-dom
import MainNavigation from './MainNavigation';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'), // use actual Link component
  Link: ({ to, children }: { to: string; children: React.ReactNode }) => (
    <a href={to}>{children}</a>
  ),
}));

test('renders MainNavigation correctly', () => {
  render(
    <MemoryRouter>
      <MainNavigation />
    </MemoryRouter>
  );

  // Use Jest's snapshot testing to ensure the rendered component structure remains consistent
  expect(screen.getByTestId('main-navigation')).toMatchSnapshot();
});
