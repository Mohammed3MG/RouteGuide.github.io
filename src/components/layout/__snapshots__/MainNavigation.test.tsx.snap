// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainNavigation correctly 1`] = `
<nav
  class="navbar navbar-expand-lg navbar-light bg-light"
  data-testid="main-navigation"
>
  <div
    class="container"
  >
    <a
      href="/"
    >
      <span
        aria-label="map"
        role="img"
      >
        üó∫Ô∏è
      </span>
       
      Travel Route guide
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      class="navbar-toggler collapsed"
      type="button"
    >
      <span
        class="navbar-toggler-icon"
      />
    </button>
    <div
      class="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        class="me-auto navbar-nav"
      >
        <a
          href="/home"
        >
          Home
        </a>
        <a
          href="/cities"
        >
          Cities
        </a>
        <a
          href="/result"
        >
          Result
        </a>
      </div>
    </div>
  </div>
</nav>
`;
